conn.go

```diff
@@ -260,10 +260,14 @@ func (c *Conn) clearPendingRequests() {
 }
 
 func (c *Conn) getXid() int32 {
+	c.xidMu.Lock()
+	defer c.xidMu.Unlock()
+
 	if c.xid == math.MaxInt32 {
 		c.xid = 1
```

Comment:
Why 1 and not 0 for example? Is this something zk specific? Maybe add some comment about it?

Reply:
After looking into it a bit more, it seems like it can be 0 as well.

Reply:
As discussed offline, we can have 0. That also mean we could simply use a bitwise mask instead of this branching. That also allows to go lock-free by using `atomic.AddInt32`.